bundle-workflow/src/aws/s3_bucket.py

```diff
@@ -99,6 +99,7 @@ def download_file(self, key, dest):
         local_dir = Path(dest)
         file_name = key.split("/")[-1]
         target = Path(local_dir) / Path(file_name)
+        target.parent.mkdir(parents=True, exist_ok=True)
```

Comment:
Feels like the caller should be ensuring that the target folder it passes into `download_file` exists.